src/main/java/com/google/devtools/build/lib/analysis/platform/ConstraintCollection.java

```diff
@@ -233,9 +233,10 @@ public Sequence<ConstraintSettingInfo> constraintSettings() {
 
   @Override
   public Object getIndex(StarlarkSemantics semantics, Object key) throws EvalException {
-    Object result = get(convertKey(key));
+    ConstraintSettingInfo constraintSettingInfo = convertKey(key);
+    Object result = get(constraintSettingInfo);
     if (result == null) {
-      result = Starlark.NONE;
```

Comment:
I am really sorry for being unclear (and for being mistaken in my first comment) - the previous version of the code you hard (returning Starlark.NONE) was correct. For consistency with `dict`'s behavior, we don't want getIndex(x) to throw if containsKey(x) returns true.

Reply:
Oh, I see. Changed it back.